runtimes/polimec/src/lib.rs

```diff
@@ -1152,6 +1158,34 @@ impl pallet_dispenser::Config for Runtime {
 	type WhitelistedPolicy = DispenserWhitelistedPolicy;
 }
 
+pub struct PLMCToFundingAssetBalance;
+impl ConversionToAssetBalance<Balance, AssetId, Balance> for PLMCToFundingAssetBalance {
+	type Error = InvalidTransaction;
+
+	fn to_asset_balance(plmc_balance: Balance, asset_id: AssetId) -> Result<Balance, Self::Error> {
+		let plmc_price =
+			<PriceProviderOf<Runtime>>::get_decimals_aware_price(PLMC_FOREIGN_ID, USD_DECIMALS, PLMC_DECIMALS)
+				.ok_or(InvalidTransaction::Payment)?;
+		let funding_asset_decimals = <ForeignAssets as fungibles::metadata::Inspect<AccountId>>::decimals(asset_id);
+		let funding_asset_price =
+			<PriceProviderOf<Runtime>>::get_decimals_aware_price(asset_id, USD_DECIMALS, funding_asset_decimals)
+				.ok_or(InvalidTransaction::Payment)?;
+		let usd_balance = plmc_price.saturating_mul_int(plmc_balance);
+		let funding_asset_balance =
+			funding_asset_price.reciprocal().ok_or(InvalidTransaction::Payment)?.saturating_mul_int(usd_balance);
+		Ok(funding_asset_balance)
```

Comment:
The logic should be good. The reciprocal always looks a bit scary, but it should be okay. Let's see how it behaves in the tests.